core/src/main/java/com/linecorp/armeria/client/retry/HttpStatusBasedRetryStrategy.java

```diff
@@ -32,7 +32,10 @@
 /**
  * Provides a {@link RetryStrategy} that decides to retry the request based on the {@link HttpStatus} of
  * its response or the {@link Exception} raised while processing the {@link HttpResponse}.
+ *
+ * @deprecated Use {@link RetryStrategy#builder()}
  */
+@Deprecated
```

Comment:
It's not part of the public API no you can just delete it when it becomes unused.